REST_based_Auth_API/my_rest_api/core/views.py

```diff
@@ -0,0 +1,371 @@
+from decimal import Context
+import json
+
+from django.db.models.expressions import OrderBy
+from django.http import request
+from django.shortcuts import render
+
+from rest_framework import generics, status, viewsets
+from rest_framework.decorators import action
+from rest_framework import generics
+from rest_framework.fields import NullBooleanField
+from rest_framework.serializers import Serializer
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework.permissions import AllowAny, IsAuthenticated
+from django.contrib.auth.models import User
+from rest_framework_simplejwt.views import TokenObtainPairView
+
+from .models import Comment, Friend, Like, Message, Notification, Post, Request
+from .serializers import (
+    CommentSerializer,
+    FriendsSerializer,
+    LikeSerializer,
+    NotificationSerializer,
+    UserSerializer,
+    PostSerializer,
+    RequestSerializer,
+    MessageSerializer,
+    ChangeRequestStatusSerializer,
+    MyTokenObtainPairSerializer,
+)
+from .utilities import create_notification, save_friend
+
+
+class MyTokenObtainPairView(TokenObtainPairView):
+    serializer_class = MyTokenObtainPairSerializer
+
+
+class HelloView(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        content = {"message": "Hello Faraz!"}
+        return Response(content)
+
+
+class RegisterUserView(generics.GenericAPIView):
+    serializer_class = UserSerializer
+    permission_classes = (AllowAny,)
+    queryset = User.objects.all()
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.serializer_class(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.save()
+        return Response(self.serializer_class(user).data)
+
+
+class GetUsersView(viewsets.ModelViewSet):
+    serializer_class = UserSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = User.objects.all()
+
+
+class PostView(viewsets.ModelViewSet):
+    serializer_class = PostSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Post.objects.all()
+
+    def get_queryset(self):
+        user = self.request.user
+        return self.queryset.filter(user=user)
+
+
+class CommentView(viewsets.ModelViewSet):
+    serializer_class = CommentSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Comment.objects.all()
+
+    def get_queryset(self):
+        post_id = self.request.data["post"]
+        comment_id = self.request.data["comment"]
+
+        if comment_id:  # checking it because it can be null
+            return self.queryset.filter(comment=comment_id)
+        return self.queryset.filter(post=post_id)
+
+    def create(self, request, *args, **kwargs):
+        signed_in_user = self.request.user.id
+        post_id = self.request.data["post"]
+        comment_id = self.request.data["comment"]
+
+        commentor = User.objects.get(pk=signed_in_user)
```

Comment:
Why did you add this redundant query for getting the user? Don't we have the user in `self.request.user`? On line 89, you pick ID and then retrieve the user again.